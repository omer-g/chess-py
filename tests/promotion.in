e2 e4
f7 f5
e4 f5
g7 g6
f5 g6
a7 a6
g6 h7
a6 a5
h7 g8
Q
0
