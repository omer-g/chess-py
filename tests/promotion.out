
        ##########################
        ######## chess-py ########
        ##########################

                    #
                   ###
                  ####
                 #### ##
                #### ####
                 ## ###
                  ####
               ##########
                  ####
                  ####
                  ####
                  ####
             ##############
             ##############

move: 'e2 e4' revert: 'r' exit: '0'

   a  b  c  d  e  f  g  h  
8 |bR|bN|bB|bQ|bK|bB|bN|bR|
7 |bP|bP|bP|bP|bP|bP|bP|bP|
6 |  |  |  |  |  |  |  |  |
5 |  |  |  |  |  |  |  |  |
4 |  |  |  |  |  |  |  |  |
3 |  |  |  |  |  |  |  |  |
2 |wP|wP|wP|wP|wP|wP|wP|wP|
1 |wR|wN|wB|wQ|wK|wB|wN|wR|

enter move
   a  b  c  d  e  f  g  h  
8 |bR|bN|bB|bQ|bK|bB|bN|bR|
7 |bP|bP|bP|bP|bP|bP|bP|bP|
6 |  |  |  |  |  |  |  |  |
5 |  |  |  |  |  |  |  |  |
4 |  |  |  |  |wP|  |  |  |
3 |  |  |  |  |  |  |  |  |
2 |wP|wP|wP|wP|  |wP|wP|wP|
1 |wR|wN|wB|wQ|wK|wB|wN|wR|

enter move
   a  b  c  d  e  f  g  h  
8 |bR|bN|bB|bQ|bK|bB|bN|bR|
7 |bP|bP|bP|bP|bP|  |bP|bP|
6 |  |  |  |  |  |  |  |  |
5 |  |  |  |  |  |bP|  |  |
4 |  |  |  |  |wP|  |  |  |
3 |  |  |  |  |  |  |  |  |
2 |wP|wP|wP|wP|  |wP|wP|wP|
1 |wR|wN|wB|wQ|wK|wB|wN|wR|

enter move
   a  b  c  d  e  f  g  h  
8 |bR|bN|bB|bQ|bK|bB|bN|bR|
7 |bP|bP|bP|bP|bP|  |bP|bP|
6 |  |  |  |  |  |  |  |  |
5 |  |  |  |  |  |wP|  |  |
4 |  |  |  |  |  |  |  |  |
3 |  |  |  |  |  |  |  |  |
2 |wP|wP|wP|wP|  |wP|wP|wP|
1 |wR|wN|wB|wQ|wK|wB|wN|wR|

enter move
   a  b  c  d  e  f  g  h  
8 |bR|bN|bB|bQ|bK|bB|bN|bR|
7 |bP|bP|bP|bP|bP|  |  |bP|
6 |  |  |  |  |  |  |bP|  |
5 |  |  |  |  |  |wP|  |  |
4 |  |  |  |  |  |  |  |  |
3 |  |  |  |  |  |  |  |  |
2 |wP|wP|wP|wP|  |wP|wP|wP|
1 |wR|wN|wB|wQ|wK|wB|wN|wR|

enter move
   a  b  c  d  e  f  g  h  
8 |bR|bN|bB|bQ|bK|bB|bN|bR|
7 |bP|bP|bP|bP|bP|  |  |bP|
6 |  |  |  |  |  |  |wP|  |
5 |  |  |  |  |  |  |  |  |
4 |  |  |  |  |  |  |  |  |
3 |  |  |  |  |  |  |  |  |
2 |wP|wP|wP|wP|  |wP|wP|wP|
1 |wR|wN|wB|wQ|wK|wB|wN|wR|

enter move
   a  b  c  d  e  f  g  h  
8 |bR|bN|bB|bQ|bK|bB|bN|bR|
7 |  |bP|bP|bP|bP|  |  |bP|
6 |bP|  |  |  |  |  |wP|  |
5 |  |  |  |  |  |  |  |  |
4 |  |  |  |  |  |  |  |  |
3 |  |  |  |  |  |  |  |  |
2 |wP|wP|wP|wP|  |wP|wP|wP|
1 |wR|wN|wB|wQ|wK|wB|wN|wR|

enter move
   a  b  c  d  e  f  g  h  
8 |bR|bN|bB|bQ|bK|bB|bN|bR|
7 |  |bP|bP|bP|bP|  |  |wP|
6 |bP|  |  |  |  |  |  |  |
5 |  |  |  |  |  |  |  |  |
4 |  |  |  |  |  |  |  |  |
3 |  |  |  |  |  |  |  |  |
2 |wP|wP|wP|wP|  |wP|wP|wP|
1 |wR|wN|wB|wQ|wK|wB|wN|wR|

enter move
   a  b  c  d  e  f  g  h  
8 |bR|bN|bB|bQ|bK|bB|bN|bR|
7 |  |bP|bP|bP|bP|  |  |wP|
6 |  |  |  |  |  |  |  |  |
5 |bP|  |  |  |  |  |  |  |
4 |  |  |  |  |  |  |  |  |
3 |  |  |  |  |  |  |  |  |
2 |wP|wP|wP|wP|  |wP|wP|wP|
1 |wR|wN|wB|wQ|wK|wB|wN|wR|

enter move
   a  b  c  d  e  f  g  h  
8 |bR|bN|bB|bQ|bK|bB|wP|bR|
7 |  |bP|bP|bP|bP|  |  |  |
6 |  |  |  |  |  |  |  |  |
5 |bP|  |  |  |  |  |  |  |
4 |  |  |  |  |  |  |  |  |
3 |  |  |  |  |  |  |  |  |
2 |wP|wP|wP|wP|  |wP|wP|wP|
1 |wR|wN|wB|wQ|wK|wB|wN|wR|

choose promotion piece (Q, R, B, N)
   a  b  c  d  e  f  g  h  
8 |bR|bN|bB|bQ|bK|bB|wQ|bR|
7 |  |bP|bP|bP|bP|  |  |  |
6 |  |  |  |  |  |  |  |  |
5 |bP|  |  |  |  |  |  |  |
4 |  |  |  |  |  |  |  |  |
3 |  |  |  |  |  |  |  |  |
2 |wP|wP|wP|wP|  |wP|wP|wP|
1 |wR|wN|wB|wQ|wK|wB|wN|wR|

enter move
